src/test/groovy/hudson/plugins/gradle/BuildScanIntegrationTest.groovy

```diff
@@ -1,43 +1,72 @@
 package hudson.plugins.gradle
 
 import hudson.model.FreeStyleProject
+import hudson.tasks.BatchFile
+import hudson.tasks.Shell
 import org.jvnet.hudson.test.CreateFileBuilder
+import org.jvnet.hudson.test.ExtractResourceSCM
 import org.jvnet.hudson.test.JenkinsRule
 import spock.lang.Unroll
 
 @Unroll
 class BuildScanIntegrationTest extends AbstractIntegrationTest {
-    
-    def 'build scan for plugin version #buildScanVersion is discovered'() {
+
+    def 'build scans for plugin version #buildScanVersion is discovered'() {
         given:
         gradleInstallationRule.gradleVersion = gradleVersion
         gradleInstallationRule.addInstallation()
         FreeStyleProject p = j.createFreeStyleProject()
-        p.buildersList.add(new CreateFileBuilder("build.gradle", """
-plugins {
-    id 'com.gradle.build-scan' version '${buildScanVersion}'
-}
-
-buildScan {
-    licenseAgreementUrl = 'https://gradle.com/terms-of-service'
-    licenseAgree = 'yes'
-}
-
-task hello { doLast { println 'Hello' } }"""))
+        p.buildersList.add(buildScriptBuilder(buildScanVersion))
+        p.buildersList.add(new Gradle(tasks: 'hello', gradleName: gradleVersion, switches: "${args} --no-daemon"))
         p.buildersList.add(new Gradle(tasks: 'hello', gradleName: gradleVersion, switches: "${args} --no-daemon"))
 
         when:
         def build = j.buildAndAssertSuccess(p)
 
         then:
         println JenkinsRule.getLog(build)
-        def action = build.getAction(BuildScanAction)
-        action.scanUrl.contains('gradle.com')
-        new URL(action.scanUrl)
+        def action = build.getAction(BuildScansAction)
+        action.scanUrls.size() == 2
+        action.scanUrls.each { new URL(it) }
 
         where:
         buildScanVersion | gradleVersion | args
-        "1.6" | "3.4" | "-Dscan"
-        "1.8" | "4.0" | "--scan"
+        "1.6"            | "3.4"         | "-Dscan"
+        "1.8"            | "4.0"         | "--scan"
+    }
+
+    def 'build scan is discovered when using non-gradle build step'() {
+        given:
+        FreeStyleProject p = j.createFreeStyleProject()
+        p.setScm(new ExtractResourceSCM(this.class.getResource('/gradle/wrapper.zip')))
```

Comment:
I would prefer not using a checked in zip file as a resource. We could pack up the checked in wrapper in the build and provide it as a resource or install Gradle via the existing tool support used in other tests.

Reply:
Good idea. The build now packages up the existing wrapper and includes in on the test classpath.